### DISCUSSION TITLE
Support for First Person View with basemap library

### DISCUSSION THREAD

**Discussion by chrisgervang** [visgl/deck.gl#5429] [Ideas]
I was wondering what would be involved in adding support for FirstPersonView with StaticMap in `react-map-gl`? I'm also not sure if this is a deck question or react-map-gl question, but I suspect both. 

Currently using FirstPersonView with StaticMap results in a `failed to invert matrix` error ([pictured here](https://github.com/visgl/deck.gl/issues/4954#issue-703943343)). 

Reviewing MapView/[WebMercatorViewport code](https://github.com/visgl/deck.gl/blob/master/modules/core/src/viewports/web-mercator-viewport.js) I see conformance to the "default" mapbox-js style viewport - is there a similar setup necessary to support the ["free camera"](https://github.com/mapbox/mapbox-gl-js/blob/main/src/ui/free_camera.js) mapbox-js style viewport?

**Comment by Pessimistress** [comment]
The issue in deck.gl x mapbox-gl interop is that deck currently passes the following parameters to Mapbox: `longtidue`, `latitude`, `zoom`, `pitch` and `bearing`. Not the actual viewport. Is there an example on how "free camera" is used in mapbox?

**Reply by chrisgervang** [reply]
Yeah, [example 1](https://docs.mapbox.com/mapbox-gl-js/example/free-camera-path/) and [example 2](https://docs.mapbox.com/mapbox-gl-js/example/free-camera-point/). The camera is set with [`map.setFreeCameraOptions`](https://docs.mapbox.com/mapbox-gl-js/api/properties/#freecameraoptions). 

I looked around the [deck react](https://github.com/visgl/deck.gl/tree/master/modules/react/src) package for where it passes the parameters (I assumed it was as props), but it seems like that isn't where the interop is happening?

**Reply by chrisgervang** [reply]
To set the free camera option one could do:
```
var coord = mapboxgl.MercatorCoordinate.fromLngLat({ lng: 0, lat: 0}, 0);
map.setFreeCameraOptions(coord, orientationQuat);
```
Where lat/lng/altitude/orientationQuat come from deck.

**Reply by Pessimistress** [reply]
This needs to be supported in react-map-gl. DeckGL passes down a `viewState` object but it doesn't specify the view that it's attached to. `StaticMap` could add a prop to switch on free camera mode.

Another issue is that Deck's first-person viewport does not match that of mapbox's. Some investigation will be required to see how to get them aligned.

**Reply by pcace** [reply]
Any news on this topic? Is there a way to get the firstperson to work in a mapbox environment?
